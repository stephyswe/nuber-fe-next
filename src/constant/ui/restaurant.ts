export const restaurantData = [
  {
    slug: 'burger-king-jarntorget',
    name: 'Burger King Järntorget',
    coverImg:
      'https://cn-geo1.uber.com/image-proc/resize/eats/format=webp/width=550/height=440/quality=70/srcb64=aHR0cHM6Ly9kMXJhbHNvZ25qbmczNy5jbG91ZGZyb250Lm5ldC9lMTIxZDlhMS01YzRiLTQ5OGEtYWM0MC1lMjZhMTlhZThhYjUuanBlZw==',
    category: ['Burgers', 'Fast Food'],
    address: 'Järntorget 6, Västra Götalands län 413 04',
    ranking: null,
    banner: 'SEK 0 Delivery Fee (Spend SEK 250)',
  },
  {
    slug: 'max-goteborg-nordstan',
    name: 'MAX Göteborg - Nordstan',
    coverImg:
      'https://cn-geo1.uber.com/image-proc/resize/eats/format=webp/width=550/height=440/quality=70/srcb64=aHR0cHM6Ly9kMXJhbHNvZ25qbmczNy5jbG91ZGZyb250Lm5ldC81YWUzN2IxZi1jOTE2LTQ1ODEtOTY5YS00NjMyYWRlMTRiNjQuanBlZw==',
    category: ['Burgers', 'Fast Food'],
    address: 'Norra Hamngatan 20, Gothenburg, Gothenburg 411 06',
    ranking: '4.0',
    banner: 'SEK 0 Delivery Fee (Spend SEK 250)',
  },

  {
    slug: 'max-goteborg-nordstan',
    name: 'Injera Restaurang and Bar',
    coverImg:
      'https://cn-geo1.uber.com/image-proc/resize/eats/format=webp/width=550/height=440/quality=70/srcb64=aHR0cHM6Ly9kMXJhbHNvZ25qbmczNy5jbG91ZGZyb250Lm5ldC82MTdkODdiNS1mOWNhLTQzZjctYmU5OC1hODU4YjQ4YTBiYzAuanBlZw==',
    category: ['Ethiopian', 'African'],
    address: 'Kobbarnas Väg 1c, Göteborg, 416 47',
    ranking: null,
    banner: null,
  },
  {
    slug: 'max-goteborg-nordstan',
    name: 'Burger King Järntorget',
    coverImg:
      'https://cn-geo1.uber.com/image-proc/resize/eats/format=webp/width=550/height=440/quality=70/srcb64=aHR0cHM6Ly9kMXJhbHNvZ25qbmczNy5jbG91ZGZyb250Lm5ldC9lMTIxZDlhMS01YzRiLTQ5OGEtYWM0MC1lMjZhMTlhZThhYjUuanBlZw==',
    category: ['Burgers', 'Fast Food'],
    address: 'Järntorget 6, Västra Götalands län 413 04',
    ranking: null,
    banner: 'SEK 0 Delivery Fee (Spend SEK 250)',
  },
  {
    slug: 'max-goteborg-nordstan',
    name: 'SOS Candy Hisingen',
    coverImg:
      'https://cn-geo1.uber.com/image-proc/resize/eats/format=webp/width=550/height=440/quality=70/srcb64=aHR0cHM6Ly9kMXJhbHNvZ25qbmczNy5jbG91ZGZyb250Lm5ldC8wMDU2Yjk3Mi05NGZkLTRjZDgtYTA5MC0yYzdjODhiOTdmMjQuanBlZw==',
    category: ['Candy', 'Snacks', 'Drinks', 'Grocery'],
    address: 'Hjalmar Brantingsgatan 5d, Göteborg, EMEA 417 06',
    ranking: null,
    banner: null,
  },
  {
    slug: 'max-goteborg-nordstan',
    name: 'Paradis Corner',
    coverImg:
      'https://cn-geo1.uber.com/image-proc/resize/eats/format=webp/width=550/height=440/quality=70/srcb64=aHR0cHM6Ly90Yi1zdGF0aWMudWJlci5jb20vcHJvZC9pbWFnZS1wcm9jL3Byb2Nlc3NlZF9pbWFnZXMvNjNlNmNiOTkyNDBjZDI4ODY2NjcxNGFiZmY1OGMxN2IvODIwODgzYTQ4NTY3NjcwYWNiZDcyMGJjNzYzOTEyOTEuanBlZw==',
    category: ['Burgers', 'Fast Food'],
    address: 'Järntorget 6, Västra Götalands län 413 04',
    ranking: null,
    banner: null,
  },
  {
    slug: 'max-goteborg-nordstan',
    name: 'SOS Candy Hisingen',
    coverImg:
      'https://cn-geo1.uber.com/image-proc/resize/eats/format=webp/width=550/height=440/quality=70/srcb64=aHR0cHM6Ly9kMXJhbHNvZ25qbmczNy5jbG91ZGZyb250Lm5ldC8wMDU2Yjk3Mi05NGZkLTRjZDgtYTA5MC0yYzdjODhiOTdmMjQuanBlZw==',
    category: ['Candy', 'Snacks', 'Drinks', 'Grocery'],
    address: 'Stampgatan 15, Göteborg, EMEA 416 64',
    ranking: null,
    banner: null,
  },
  {
    slug: 'max-goteborg-nordstan',
    name: 'MAX Göteborg - Nordstan',
    coverImg:
      'https://cn-geo1.uber.com/image-proc/resize/eats/format=webp/width=550/height=440/quality=70/srcb64=aHR0cHM6Ly9kMXJhbHNvZ25qbmczNy5jbG91ZGZyb250Lm5ldC81YWUzN2IxZi1jOTE2LTQ1ODEtOTY5YS00NjMyYWRlMTRiNjQuanBlZw==',
    category: ['Burgers', 'Fast Food'],
    address: 'Norra Hamngatan 20, Gothenburg, Gothenburg 411 06',
    ranking: '4.0',
    banner: 'SEK 0 Delivery Fee (Spend SEK 250)',
  },
];

export const restaurantTitleData = [
  {
    title: 'Today’s offers',
    items: [
      {
        slug: 'burger-king-jarntorget',
        name: 'Burger King Järntorget',
        coverImg:
          'https://cn-geo1.uber.com/image-proc/resize/eats/format=webp/width=550/height=440/quality=70/srcb64=aHR0cHM6Ly9kMXJhbHNvZ25qbmczNy5jbG91ZGZyb250Lm5ldC9lMTIxZDlhMS01YzRiLTQ5OGEtYWM0MC1lMjZhMTlhZThhYjUuanBlZw==',
        category: ['Burgers', 'Fast Food'],
        address: 'Järntorget 6, Västra Götalands län 413 04',
        ranking: null,
        banner: 'SEK 0 Delivery Fee (Spend SEK 250)',
      },
      {
        slug: 'max-goteborg-nordstan',
        name: 'MAX Göteborg - Nordstan',
        coverImg:
          'https://cn-geo1.uber.com/image-proc/resize/eats/format=webp/width=550/height=440/quality=70/srcb64=aHR0cHM6Ly9kMXJhbHNvZ25qbmczNy5jbG91ZGZyb250Lm5ldC81YWUzN2IxZi1jOTE2LTQ1ODEtOTY5YS00NjMyYWRlMTRiNjQuanBlZw==',
        category: ['Burgers', 'Fast Food'],
        address: 'Norra Hamngatan 20, Gothenburg, Gothenburg 411 06',
        ranking: '4.0',
        banner: 'SEK 0 Delivery Fee (Spend SEK 250)',
      },
    ],
  },
  {
    title: 'Popular near you',
    items: [
      {
        slug: 'max-goteborg-nordstan',
        name: 'Injera Restaurang and Bar',
        coverImg:
          'https://cn-geo1.uber.com/image-proc/resize/eats/format=webp/width=550/height=440/quality=70/srcb64=aHR0cHM6Ly9kMXJhbHNvZ25qbmczNy5jbG91ZGZyb250Lm5ldC82MTdkODdiNS1mOWNhLTQzZjctYmU5OC1hODU4YjQ4YTBiYzAuanBlZw==',
        category: ['Ethiopian', 'African'],
        address: 'Kobbarnas Väg 1c, Göteborg, 416 47',
        ranking: null,
        banner: null,
      },
      {
        slug: 'max-goteborg-nordstan',
        name: 'Burger King Järntorget',
        coverImg:
          'https://cn-geo1.uber.com/image-proc/resize/eats/format=webp/width=550/height=440/quality=70/srcb64=aHR0cHM6Ly9kMXJhbHNvZ25qbmczNy5jbG91ZGZyb250Lm5ldC9lMTIxZDlhMS01YzRiLTQ5OGEtYWM0MC1lMjZhMTlhZThhYjUuanBlZw==',
        category: ['Burgers', 'Fast Food'],
        address: 'Järntorget 6, Västra Götalands län 413 04',
        ranking: null,
        banner: 'SEK 0 Delivery Fee (Spend SEK 250)',
      },
      {
        slug: 'max-goteborg-nordstan',
        name: 'Burger King Järntorget',
        coverImg:
          'https://cn-geo1.uber.com/image-proc/resize/eats/format=webp/width=550/height=440/quality=70/srcb64=aHR0cHM6Ly9kMXJhbHNvZ25qbmczNy5jbG91ZGZyb250Lm5ldC9lMTIxZDlhMS01YzRiLTQ5OGEtYWM0MC1lMjZhMTlhZThhYjUuanBlZw==',
        category: ['Burgers', 'Fast Food'],
        address: 'Järntorget 6, Västra Götalands län 413 04',
        ranking: null,
        banner: 'SEK 0 Delivery Fee (Spend SEK 250)',
      },
    ],
  },
  {
    title: 'Healthy eating',
    items: [
      {
        slug: 'max-goteborg-nordstan',
        name: 'Injera Restaurang and Bar',
        coverImg:
          'https://cn-geo1.uber.com/image-proc/resize/eats/format=webp/width=550/height=440/quality=70/srcb64=aHR0cHM6Ly9kMXJhbHNvZ25qbmczNy5jbG91ZGZyb250Lm5ldC82MTdkODdiNS1mOWNhLTQzZjctYmU5OC1hODU4YjQ4YTBiYzAuanBlZw==',
        category: ['Ethiopian', 'African'],
        address: 'Kobbarnas Väg 1c, Göteborg, 416 47',
        ranking: null,
        banner: null,
      },
      {
        slug: 'max-goteborg-nordstan',
        name: 'Burger King Järntorget',
        coverImg:
          'https://cn-geo1.uber.com/image-proc/resize/eats/format=webp/width=550/height=440/quality=70/srcb64=aHR0cHM6Ly9kMXJhbHNvZ25qbmczNy5jbG91ZGZyb250Lm5ldC9lMTIxZDlhMS01YzRiLTQ5OGEtYWM0MC1lMjZhMTlhZThhYjUuanBlZw==',
        category: ['Burgers', 'Fast Food'],
        address: 'Järntorget 6, Västra Götalands län 413 04',
        ranking: null,
        banner: 'SEK 0 Delivery Fee (Spend SEK 250)',
      },
    ],
  },
];
