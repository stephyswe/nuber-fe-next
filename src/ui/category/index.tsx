import clsx from 'clsx';
import { useRouter } from 'next/router';

import { useWindowSizeJs } from '@/hooks/useWindowSizeJs';

//TODO: Fix this type
// eslint-disable-next-line unused-imports/no-unused-vars
type CategoriesProps = CategoryItemSquareProps[] | null | undefined;

export const CategoryList = ({ data }: { data: any }) => {
  const { isMobile } = useWindowSizeJs();
  return (
    <div
      data-testid='ui-category-list'
      className='mr-6 block w-full flex-shrink-0 flex-grow-0 basis-full md:scroll-align-start'
    >
      <div className='-mx-3 -mb-6 box-border flex flex-wrap'>
        {data
          ?.slice(0, isMobile ? 4 : 12)
          .map((item: CategoryItemSquareProps) => (
            <CategorySquareItem key={item.id} {...item} />
          ))}
      </div>
    </div>
  );
};

/* eslint-disable @next/next/no-img-element */
export const CategorySquareItem = ({
  name,
  coverImg,
  slug,
}: CategoryItemSquareProps) => {
  const router = useRouter();
  return (
    <div className='ml-0 mb-6 box-border block w-[50%] p-[0px_6px] md:relative md:w-[16.666666666666668%] md:flex-none md:px-3'>
      <a
        className='relative block h-[65px] bg-[#f6f0ea] md:h-20 md:overflow-hidden '
        href={`/client/category/${router.query.label}/${slug}`}
      >
        <img
          className='absolute right-0 top-0 h-full'
          alt={name}
          src={coverImg ? coverImg : ''}
          aria-hidden='true'
        />
        <div
          className={clsx(
            'relative box-border min-w-[120px] p-4 text-base font-medium leading-6 text-black md:w-[60%] md:text-lg md:leading-6'
          )}
        >
          <div className='overflow-hidden webkit-line-clamp-2 webkit-orient-vertical webkit'>
            {name}
          </div>
        </div>
      </a>
    </div>
  );
};

export type CategoryItemSquareProps = {
  __typename?: 'Category' | undefined;
  id: number;
  name: string;
  coverImg?: string | null | undefined;
  slug: string;
  countRestaurants: number;
};
