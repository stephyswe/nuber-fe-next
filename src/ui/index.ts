export { Accordion } from './accordion';
export type { BreadCrumbItemProps } from './breadcrumb';
export { BreadCrumb } from './breadcrumb';
export { Cart } from './cart';
export { CategoryList, CategorySquareItem } from './category';
export { Container } from './container';
export { Headline, HeadlineNavigate } from './headline';
export { Hero, HeroDynamic } from './hero';
export * from './loading';
export { Logo } from './logo';
export { HomeMap, MapButton } from './maps';
export {
  DishModal,
  ModalAllergy,
  ModalBuy,
  ModalDetail,
  ModalHeader,
  ModalImage,
} from './modals';
export { Separator, Spacer, SpacerItem } from './spacer';

/* icons has separate export */
