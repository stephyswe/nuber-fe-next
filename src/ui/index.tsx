import clsx from 'clsx';

import { useWindowSizeJs } from '@/hooks/useWindowSizeJs';

export type { BreadCrumbItemProps } from './breadcrumb';
export { BreadCrumb } from './breadcrumb';
export { Cart } from './cart';
export { Container } from './container';
export { Headline, HeadlineNavigate } from './headline';
export { DynamicHero, Hero } from './hero';
export { Toggler } from './toggler';

/* maps */
export * from './loading';
export { HomeMap, MapButton } from './maps';

type SpacerProps = {
  length?: number;
  index?: number;
} & React.ComponentPropsWithRef<'div'>;

export const Spacer = ({ length, index, className, ...rest }: SpacerProps) => {
  if (length && index) if (length === index + 1) return null;
  return (
    <div
      className={clsx(
        'm-0 flex-shrink-0 p-0',
        className && className.charAt(0) !== 'h' ? 'h-[1px]' : '',
        className
      )}
      {...rest}
    ></div>
  );
};

export const SpacerItem = ({ length, index, children }: SpacerProps) => {
  if (length && index) if (length === index + 1) return null;
  return <>{children}</>;
};

export const NewLogo = () => (
  // eslint-disable-next-line @next/next/no-img-element
  <img
    alt='Startsida fÃ¶r Uber&nbsp;Eats'
    src='/images/logo.svg'
    width='146'
    height='24'
    className='block h-6 max-h-[18px] w-auto max-w-[136px] md:max-h-[inherit] md:max-w-[inherit] '
  />
);

export const Separator = ({
  topOnly,
  mobileHidden,
  mobileHiddenSpace,
}: any) => {
  const { isMobile } = useWindowSizeJs();
  return (
    <hr
      className={clsx(
        'h-[1px] border-none bg-[#e2e2e2]',
        isMobile && mobileHidden ? 'hidden' : '',
        isMobile && mobileHiddenSpace ? 'my-4 bg-[transparent]' : '',
        topOnly ? 'm-[32px_0_0]' : 'my-8'
      )}
    />
  );
};
