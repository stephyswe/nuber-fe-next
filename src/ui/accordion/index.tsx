import clsx from 'clsx';
import React, { Fragment, useEffect, useRef, useState } from 'react';

import { Button } from '@/components';

import { useOrders } from '@/contexts';
import { Spacer, SpacerItem } from '@/ui';
import { AccordionItem } from '@/ui/accordion/item';
import { onClick } from '@/ui/accordion/utils';
import { SvgDropdown } from '@/ui/icons';

import { AccordionProps, AccordionTopProps } from './types';

export const Accordion = ({ data, dishId }: AccordionProps) => {
  const { name, choices } = data;
  const { setOrderItem } = useOrders();
  const [isOpened, setOpened] = useState<boolean>(false);
  const [height, setHeight] = useState<string>('0px');
  const contentRef = useRef<HTMLDivElement>(null);
  const [isActive, setActive] = useState<string>('');

  useEffect(() => {
    onToggle();
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, []);

  const onToggle = () => {
    setOpened(!isOpened);
    setHeight(!isOpened ? `${contentRef?.current?.['scrollHeight']}px` : '0px');
  };

  const handleClick = (title: string) => {
    onClick(title, choices, setOrderItem, setActive, isActive, dishId, name);
  };

  return (
    <div>
      <AccordionTop name={name} onToggle={onToggle} isOpened={isOpened} />
      <div
        ref={contentRef}
        style={{ height: height }}
        className={clsx(isOpened ? 'p-[24px_16px]' : 'opacity-0', '')}
      >
        <form>{checkRender(choices, dishId, handleClick, isActive, name)}</form>
      </div>
    </div>
  );
};

function checkRender(
  choices: AccordionProps['data']['choices'],
  dishId: number,
  onClick: (title: string) => void,
  isActive: string,
  name: string
) {
  if (choices) {
    return (
      <>
        {choices.map((choice: { name: string }, index: number) => (
          <Fragment key={index}>
            <AccordionItem
              title={choice.name}
              dishId={dishId}
              onClick={onClick}
              isActive={choice.name === isActive}
            />
            <SpacerItem index={index} length={choices.length}>
              <Spacer className='mt-6' />
            </SpacerItem>
          </Fragment>
        ))}
      </>
    );
  } else {
    return (
      <AccordionItem
        type='checkbox'
        title={name}
        onClick={onClick}
        isActive={name === isActive}
      />
    );
  }
}

export const AccordionTop = ({
  name,
  onToggle,
  isOpened,
}: AccordionTopProps) => (
  <div className='flex flex-row items-center justify-between bg-[#f6f6f6] p-4'>
    <div className='flex flex-col'>
      <div className='text-lg font-medium leading-6'>Choose {name}</div>
      <div className='text-sm font-normal text-[#545454]'>
        <span>{name === 'Pickle' ? 'Choose up to 1' : 'Required'}</span>
      </div>
    </div>
    <div className='flex flex-row'>
      <Spacer className='w-2' />
      <Button
        aria-label='ui-accordion-btn-close'
        onClick={onToggle}
        className='flex h-[30px] w-[30px] items-center justify-center rounded-[50%] border-[1px] border-solid border-[#e2e2e2] bg-white'
      >
        <div className='h-6 w-6 leading-[1px]'>
          <SvgDropdown rotate={isOpened ? true : false} />
        </div>
      </Button>
    </div>
  </div>
);
