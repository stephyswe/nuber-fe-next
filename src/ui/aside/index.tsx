/* eslint-disable unused-imports/no-unused-vars */

import { Button, Link } from '@/components/';

import { Spacer } from '@/ui';

type StickyItemProps = { title: string };

export const AsideList = ({ data }: { data: any }) => (
  <div className='sticky top-20 flex h-[300px] max-w-full flex-shrink-0 overflow-y-auto'>
    <div className='w-[306px] flex-1'>
      <Spacer className='h-[24px]' />
      <nav className='flex flex-1 flex-col'>
        {Object.keys(data).map((key: string) => (
          <AsideItem key={key} title={key} />
        ))}
      </nav>
    </div>
  </div>
);

const AsideItem = ({ title }: StickyItemProps) => (
  <div className='flex items-center justify-between'>
    <Link href={`#${title}`}>
      <Button className='mb-6'>
        <div className='text-base font-medium leading-5'>{title}</div>
      </Button>
    </Link>
  </div>
);
