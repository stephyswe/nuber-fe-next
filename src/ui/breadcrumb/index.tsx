import { Fragment } from 'react';

import { Link } from '@/components';

import { SpacerItem } from '@/ui/spacer';

import { SvgBreadCrumb } from '../icons';

export type BreadCrumbItemProps = {
  title: string;
  link: string;
};

export const BreadCrumb = ({ data }: { data: BreadCrumbItemProps[] }) => (
  <ol
    data-testid='ui-breadcrumb'
    aria-label='ui-bc-list'
    className='m-[0px_0px_16px] flex items-center overflow-auto whitespace-nowrap p-0 text-[#afafaf]'
  >
    {data.map((item: BreadCrumbItemProps, index: number) => (
      <Fragment key={index}>
        <BreadCrumbItem key={item.title} title={item.title} link={item.link} />
        <SpacerItem length={data.length} index={index}>
          <div className='w-2' />
          <SvgBreadCrumb />
          <div className='w-2' />
        </SpacerItem>
      </Fragment>
    ))}
  </ol>
);

const BreadCrumbItem = ({ title, link }: BreadCrumbItemProps) => (
  <li className='inline text-sm font-medium leading-4 last-of-type:text-[#000]'>
    <Link aria-label='ui-bc-link' href={link}>
      {title}
    </Link>
  </li>
);
