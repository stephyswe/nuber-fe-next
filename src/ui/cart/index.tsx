import clsx from 'clsx';
import router from 'next/router';
import { useState } from 'react';

import { Button } from '@/components';
import { ButtonSizeProps } from '@/components/buttons/UnStyledButton';

import { useCreateOrderMutation } from '@/__generated__/graphql';
import { isTest } from '@/constant/env';
import { useDelivery, useOrders } from '@/contexts';
import { Spacer } from '@/ui';
import { CartWindow } from '@/ui/cart/list';
import { SvgCart } from '@/ui/icons';
import { LoadingInit, LoadingRound } from '@/ui/loading/item';

export const Cart = ({ size }: { size?: ButtonSizeProps }) => {
  const [showCart, setShowCart] = useState<boolean>(false);
  const { isComplete } = useDelivery();
  const { orderItems } = useOrders();

  const [createOrderMutation, { loading: placingOrder }] =
    useCreateOrderMutation({
      onCompleted(data: any) {
        const {
          createOrder: { orderId },
        } = data;
        if (data.createOrder.ok) {
          router.push(`/client/order/${orderId}`);
        }
      },
    });

  const triggerConfirmOrder = () => {
    if (placingOrder) {
      return;
    }
    if (orderItems.length === 0) {
      alert("Can't place empty order");
      return;
    }

    createOrderMutation({
      variables: { input: { restaurantId: 1, items: orderItems } },
    });
  };

  if (!isComplete && !isTest) {
    return (
      <LoadingInit w='300' h='48'>
        <LoadingRound w='80' />
      </LoadingInit>
    );
  }

  function onClick() {
    setShowCart(!showCart);
  }
  return (
    <>
      <div className='flex flex-grow-0 items-center justify-end'>
        <div className='relative flex max-w-[300px] opacity-100 transition-width-opacity-400'>
          <div className='flex cursor-auto flex-row items-stretch overflow-hidden whitespace-nowrap rounded-[500px] bg-black text-base font-medium leading-5'>
            <Button onClick={onClick} variant='btnCart' size={size}>
              <SvgCart size={size} />
              <Spacer className='w-2' />
              <div
                className={clsx(
                  'hidden md:block',
                  size === 'small' ? 'text-sm' : 'text-base'
                )}
              >
                Cart â€¢ {orderItems.length}
              </div>
            </Button>
          </div>
        </div>
      </div>
      <CartWindow
        onClose={setShowCart}
        show={showCart}
        orderItems={orderItems}
        onBuy={triggerConfirmOrder}
      />
    </>
  );
};
