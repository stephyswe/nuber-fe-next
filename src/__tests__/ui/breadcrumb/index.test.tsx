import { render, screen, takeSnapshot } from 'test-utils';

import { BreadCrumb } from '@/ui';

// ** Snapshot ** //

takeSnapshot(<BreadCrumb data={[]} />);

// ** Default ** //

test('container has html elements', () => {
  render(<BreadCrumb data={[]} />);
  const list = screen.getByRole('list', {
    name: 'ui-bc-list',
  });
  expect(list).toBeInTheDocument();
});

// ** Data ** //

test('container with one data item', () => {
  render(<BreadCrumb data={[{ title: 'Home', link: '/' }]} />);
  const link = screen.getByRole('link');
  expect(link).toBeInTheDocument();
  expect('Home').toBe(link.textContent);
  expect('/').toBe(link.getAttribute('href'));
});
