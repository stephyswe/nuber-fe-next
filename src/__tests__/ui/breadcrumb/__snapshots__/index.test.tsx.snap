// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<ol
  aria-label="ui-bc-list"
  className="m-[0px_0px_16px] flex items-center overflow-auto whitespace-nowrap p-0 text-[#afafaf]"
  data-testid="ui-breadcrumb"
/>
`;
